api_app/api/dependencies/database.py

```diff
@@ -28,8 +28,8 @@ async def _connect_to_db(cls) -> CosmosClient:
         logger.debug(f"Connecting to {STATE_STORE_ENDPOINT}")
 
         credential = await get_credential_async()
-        if MANAGED_IDENTITY_CLIENT_ID:
-            logger.debug("Connecting with managed identity")
+        if MANAGED_IDENTITY_CLIENT_ID or ENABLE_LOCAL_DEBUGGING:
```

Comment:
This allows local entra ID authentication to the CosmosDB instance (once the appropriate firewall rules are added)